x/concentrated-liquidity/types/params.go

```diff
@@ -149,3 +157,40 @@ func validateBalancerSharesDiscount(i interface{}) error {
 
 	return nil
 }
+
+// validateAuthorizedUptimes validates a slice of authorized uptimes for a given pool.
+//
+// Parameters:
+// - i: The parameter to validate.
+//
+// Returns:
+// - An error if given type is not duration slice.
+// - An error if given slice is empty.
+// - An error if any of the uptimes are invalid (i.e. not part of the list of supported uptimes).
+func validateAuthorizedUptimes(i interface{}) error {
+	authorizedUptimes, ok := i.([]time.Duration)
+
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	if len(authorizedUptimes) == 0 {
+		return fmt.Errorf("authorized uptimes cannot be empty")
+	}
+
+	// Check if each passed in uptime is in the list of supported uptimes
+	for _, uptime := range authorizedUptimes {
+		supported := false
+		for _, supportedUptime := range SupportedUptimes {
+			if uptime == supportedUptime {
+				supported = true
+			}
```

Comment:
I think we can also break here to save on unnecessary iterations.

Reply:
4060398207ce4038a9ab1307c30180f62b75aa17